### PR TITLE
Add support for dynamic notice & comment homepage

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**

### CODE REVIEW

**Comment by tadhg-ohiggins** [comment]
Supports https://github.com/eregs/notice-and-comment/pull/365

File: regulations/views/notice_home.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from operator import itemgetter
```
Comment (cmc333333) [inline regulations/views/notice_home.py]: Very neat!

File: regulations/views/notice_home.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+
+class NoticeHomeView(View):
+    """
+    Basic view that provides a list of regulations and notices to the context.
+    """
+    template_name = None  # We should probably have a default notice template.
```
Comment (cmc333333) [inline regulations/views/notice_home.py]: Agreed

File: regulations/views/notice_home.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    _, meta, _ = notice_data(notice["document_number"])
+                    notices_meta.append(meta)
+            except Http404:
+                pass
+
+        notices_meta = sorted(notices_meta, key=itemgetter("publication_date"),
+                              reverse=True)
+
+        context["notices"] = notices_meta
+        # Django templates won't show contents of CommentState as an Enum, so:
```
Comment (cmc333333) [inline regulations/views/notice_home.py]: Yeah, I [was](https://github.com/eregs/regulations-site/pull/382/commits/36c2fc9b4ba110c94a4be52ab9556a2bb3661afa) torn on this. The problem is that Django templates will silently fail if there's a typo in the `CommentState` value, so they're not getting us anything over a string comparison. The one benefit to a string comparison is not needing to add it to the template's context. That could also be accomplished by including it as a global template var, though?

I don't know the right answer, but we should be consistent.

File: regulations/views/notice_home.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        context["comment_state"] = {state.name: state.value for state in
+                                    CommentState}
+
+        assert self.template_name
+        template = self.template_name
+        return TemplateResponse(request=request, template=template,
+                                context=context)
```
Comment (cmc333333) [inline regulations/views/notice_home.py]: Have you seen Django's [TemplateView](https://docs.djangoproject.com/en/1.9/ref/class-based-views/base/#django.views.generic.base.TemplateView)? I think it does what you want, including raising an exception if no template name is given.

**Comment by cmc333333** [comment]
Two small suggestions above, @tadhg-ohiggins:
* Using TemplateView
* standardizing enum comparisons in templates

If you don't have time to address, we can circle back later; lmk

**Comment by cmc333333** [comment]
Made an issue for the enums: #400; let's not worry about TemplateView

**Pull request merged**